include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -56,7 +56,13 @@ auto /* see _Size inside the function */
 __pattern_walk_n(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Function __f, _Ranges&&... __rngs)
 {
     using _Size = std::make_unsigned_t<std::common_type_t<oneapi::dpl::__internal::__difference_t<_Ranges>...>>;
-    const _Size __n = std::min({_Size(__rngs.size())...});
+#if _ONEDPL_CPP20_RANGES_PRESENT
```

Comment:
I propose to replace this problem code (right now we have compile errors here) by `__min_size_calc::operator()` call

Reply:
```C++
const _Size __n = _Size{oneapi::dpl::__ranges::__min_size_calc{}(__rngs...)};
```

Reply:
Or you can keep explicit conversion.

Specifying the type of `min` as the template parameter is good when you know that the type is sufficient to hold all the values - e.g. if it is defined as a common type. Otherwise, explicit cast should be used. In this case, there is not much syntactic difference because of the variadic pack.